### PR TITLE
Feature gate serialization

### PR DESCRIPTION
**Pull request by messense**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,10 +54,11 @@ fn main() {
 ## Enabling Additional Features
 
 * `searcher` feature enables searcher for maximum forward matcher
+* `serialization` feature enables save and load serialized `DoubleArrayTrie` data
```
Comment (andelf) [inline README.md]: saving and loading

File: src/lib.rs [hunk 2]
```diff
@@ -47,36 +47,39 @@
 //! ## Enabling Additional Features
 //!
 //! * `searcher` feature enables searcher for maximum forward matcher
+//! * `serialization` feature enables save and load serialized `DoubleArrayTrie` data
```
Comment (andelf) [inline src/lib.rs]: ditto

File: src/lib.rs [hunk 3]
```diff
@@ -442,9 +450,12 @@ impl DoubleArrayTrie {
 #[cfg(test)]
 mod tests {
     use super::*;
+    #[cfg(feature = "serialization")]
     use std::fs::File;
+    #[cfg(feature = "serialization")]
```
Comment (andelf) [inline src/lib.rs]: Is

    #[cfg(feature = "serialization")]
    use std::{
        fs::File;
        io::BufReader;
    }

better?

Comment (messense) [reply src/lib.rs]: I think so, addressed.

**Comment by andelf** [comment]
requires `cargo fmt`.

**Comment by messense** [comment]
> requires `cargo fmt`.

Done.

File: README.md [hunk 4]
```diff
@@ -69,7 +70,7 @@ time cargo test -- --nocapture --ignored test_dat_basic
 
 ## To run benchmark tests
 ```bash
-cargo bench -- --nocapture
+cargo bench --all-features -- --nocapture
```
Comment (andelf) [inline README.md]: remove ``-- --nocapture``

Comment (andelf) [reply README.md]:
```
error: Found argument '--nocapture' which wasn't expected, or isn't valid in this context

USAGE:
    darts_benchmark-5e8c53dc2a9631b8 [FLAGS] [OPTIONS] [FILTER]

For more information try --help
```

**Pull request merged**